src/main/java/com/github/tomakehurst/wiremock/common/filemaker/FilenameMaker.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2023 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.common.filemaker;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+import com.github.tomakehurst.wiremock.extension.responsetemplating.TemplateEngine;
+import com.github.tomakehurst.wiremock.matching.AnythingPattern;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import com.github.tomakehurst.wiremock.stubbing.StubMapping;
+import java.net.URI;
+import java.text.Normalizer;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+
+public class FilenameMaker {
+  private static final Pattern NON_ALPHANUMERIC = Pattern.compile("[^\\w-]");
+  private static final Pattern WHITESPACE = Pattern.compile("[\\s]");
+  private static final String DEFAULT_FILENAME_TEMPLATE = "{{{method}}}-{{{path}}}-{{{id}}}.json";
+
+  private final TemplateEngine templateEngine;
+  private final String filenameTemplate;
+
+  public FilenameMaker(TemplateEngine templateEngine, String filenameTemplate) {
+    this.templateEngine = templateEngine;
+    this.filenameTemplate = filenameTemplate;
+  }
+
+  public FilenameMaker(TemplateEngine templateEngine) {
+    this.templateEngine = templateEngine;
+    this.filenameTemplate = DEFAULT_FILENAME_TEMPLATE;
+  }
+
+  public String filenameFor(StubMapping stubMapping) {
+    FilenameTemplateModel model = new FilenameTemplateModel(stubMapping);
+    String safeFilename = makeSafeFileName(model, "json");
+    return templateEngine.getUncachedTemplate(filenameTemplate).apply(safeFilename);
```

Comment:
2 things here:
1) We need to render the template first, then sanitise the result (so that we're sure it's a valid filename).
2) The default structure should come from a default template string, rather than replicating the string concatenation logic from the old code.